sklearn/feature_extraction/tests/test_text.py

```diff
@@ -1288,3 +1288,31 @@ def analyzer(doc):
 
     with pytest.raises(Exception, match="testing"):
         Estimator(analyzer=analyzer, input='file').fit_transform([f])
+
+
+@pytest.mark.parametrize(
+    'Vectorizer', [CountVectorizer]
+)
+def test_unused_parameters_warn(Vectorizer):
+    # adding warning messages for  parameters to check
+    msg_stop_words = "The parameter 'stop_words' will not be used"
+    " since analyzer != 'word'"
+    msg_tokenizer = "The parameter 'tokenizer' will not be used"
+    " since analyzer != 'word'"
+    msg_token_pattern = "The parameter 'token_pattern' will not be used"
+    " since 'tokenizer' is not None'"
+    vect = Vectorizer()
+    train_data = JUNK_FOOD_DOCS
+    # setting parameter and checking for corresponding warning messages
+    vect.set_params(stop_words=["you've", "you", "you'll", 'AND'],
+                    analyzer='char')
+    with pytest.warns(UserWarning, match=msg_stop_words):
+        vect.fit(train_data)
+    vect.set_params(tokenizer=lambda s: s.split(),
+                    analyzer='char')
```

Comment:
You are re-using parameters from previous modified instae, let's create a new instance at each time,
```py
vect = Vectorizer()
vect.set_params(...)
msg = '...'
with pytest.warns(UserWarning, match=msg):
   vect.fit(train_data)

etc. 
```
it's much easier to read if message is define the line above from where it's used.

Reply:
Done